DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/CentreConfiguration/RegistrationPromptsControllerTests.cs

```diff
@@ -172,13 +184,162 @@ public void AddRegistrationPromptSelectPrompt_loads_existing_temp_data()
+            registrationPromptsController.TempData.Set(initialTempData);
+            var model = new RegistrationPromptAnswersViewModel
+            {
+                Answer = null,
+                OptionsString = "Test"
+            };
+            const string action = "next";
+
+            // When
+            var result = registrationPromptsController.AddRegistrationPromptConfigureAnswers(model, action);
+
+            // Then
+            registrationPromptsController.TempData.Peek<AddRegistrationPromptData>().Should().NotBeNull();
+            registrationPromptsController.TempData.Peek<AddRegistrationPromptData>()!.SelectPromptViewModel.Should()
+                .BeEquivalentTo(expectedPromptModel);
+            registrationPromptsController.TempData.Peek<AddRegistrationPromptData>()!.ConfigureAnswersViewModel.Should()
+                .BeEquivalentTo(model);
+            result.Should().BeRedirectToActionResult().WithActionName("Index");
+        }
+
+        [Test]
+        public void AddRegistrationPromptConfigureAnswers_add_configures_new_answer_and_updates_temp_data()
+        {
+            // Given
+            var model = new RegistrationPromptAnswersViewModel
+            {
+                Answer = "Answer",
+                OptionsString = "Test"
+            };
+            var expectedPromptModel = new AddRegistrationPromptSelectPromptViewModel
+            {
+                CustomPromptId = 1,
+                Mandatory = true
+            };
+            var initialTempData = new AddRegistrationPromptData
+                { SelectPromptViewModel = expectedPromptModel, ConfigureAnswersViewModel = model };
+            registrationPromptsController.TempData.Set(initialTempData);
+            var expectedViewModel = new RegistrationPromptAnswersViewModel
+            {
+                OptionsString = "Test\r\nAnswer"
+            };
+            const string action = "addPrompt";
+
+            // When
+            var result = registrationPromptsController.AddRegistrationPromptConfigureAnswers(model, action);
+
+            // Then
+            using (new AssertionScope())
+            {
+                registrationPromptsController.TempData.Peek<AddRegistrationPromptData>().Should().NotBeNull();
+                registrationPromptsController.TempData.Peek<AddRegistrationPromptData>()!.SelectPromptViewModel.Should()
+                    .BeEquivalentTo(expectedPromptModel);
+                registrationPromptsController.TempData.Peek<AddRegistrationPromptData>()!.ConfigureAnswersViewModel
+                    .Should().BeEquivalentTo(expectedViewModel);
+                result.Should().BeViewResult();
+                result.As<ViewResult>().Model.Should().BeOfType<RegistrationPromptAnswersViewModel>();
+                result.As<ViewResult>().Model.As<RegistrationPromptAnswersViewModel>().Options.Count.Should().Be(2);

```

Comment:
We could probably benefit from pulling some of these into private methods that can be used in these tests to make things easier to read and follow, e.g. AssertNumberOfConfiguredAnswersOnView(numberOfConfiguredAnswers) or something similar
```
result.Should().BeViewResult();
result.As<ViewResult>().Model.Should().BeOfType<RegistrationPromptAnswersViewModel>();
result.As<ViewResult>().Model.As<RegistrationPromptAnswersViewModel>().Options.Count.Should().Be(2);
```